pkg/image/fetcher.go

```diff
@@ -110,9 +110,12 @@ func (f *Fetcher) Fetch(ctx context.Context, name string, options FetchOptions)
 
 	f.logger.Debugf("Pulling image %s", style.Symbol(name))
 	if err = f.pullImage(ctx, name, options.Platform); err != nil {
-		// sample error from docker engine:
-		// image with reference <image> was found but does not match the specified platform: wanted linux/amd64, actual: linux
-		if strings.Contains(err.Error(), "does not match the specified platform") {
+		// FIXME: this matching is brittle and the fallback should be removed when https://github.com/buildpacks/pack/issues/2079
+		// has been fixed for a sufficient amount of time.
+		// Sample error from docker engine:
+		// `image with reference <image> was found but does not match the specified platform: wanted linux/amd64, actual: linux`
+		if strings.Contains(err.Error(), "does not match the specified platform") &&
+			strings.HasSuffix(strings.TrimSpace(err.Error()), "actual: linux") {
```

Comment:
I added this to make the acceptance test work, but I think it's actually closer to what we want anyway